src/components/user/Team.jsx

```diff
@@ -1,46 +1,144 @@
+import Button from "../Button";
 import Input from "../Input";
+import { useEffect, useState } from "react";
+import axios from "axios";
+import toast from "react-hot-toast";
+import Loading from "../Loading";
+const Team = ({ user, setUser }) => {
+  const [team, setTeam] = useState(null);
+  const [teamID, setTeamID] = useState({});
+  const [edit, setEdit] = useState(false);
+  const handleLeave = () => {
+    axios.delete("/api/members").then(() => {
+      toast("✅ Successfully leave team!");
+      setTeam(null);
+      setUser({ ...user, team: null });
+      setTeamID({});
+    });
+  };
+  const handleJoin = () => {
+    axios
+      .put("/api/members", { team: teamID.team })
+      .then(() => {
+        toast("✅ Successfully join the team!");
+        setUser({ ...user, team: teamID.team });
+      })
+      .catch(({ response }) => {
+        if (response.data.message === "Excceed 4 People Limit")
+          toast("❌ Excceed 4 People Limit");
+        else if (response.data.message === "Invalid Team ID")
+          toast("❌ Invalid Team ID");
+        else toast("❌ Internal Server Error");
+      });
+  };
+  const handleCreate = () => {
+    axios.post("/api/team").then((response) => {
+      setTeam({
+        name: "No Team Name",
+        github: "No Link",
+        devpost: "No Link",
+        figma: "No Link",
+        members: [{ email: user.email, name: user.name }],
+      });
+      setUser({ ...user, team: response.data.items.team });
+      toast("✅ Successfully create a new team!");
+      setEdit(false);
+    });
+  };
+  const handleEdit = async () => {
+    setEdit(true);
+  };
 
-const Team = ({ user, team, setUser }) => {
+  const handleSave = () => {
+    axios.put("/api/team", team).then(() => {
+      toast("✅ Successfully Update!");
+      setEdit(false);
+    });
+  };
+  useEffect(() => {
+    user.team &&
+      axios.get("/api/team").then((response) => setTeam(response.data.items));
+  }, [user.team]);
   return (
-    <div className="bg-white rounded-lg p-4 gap-3 m-2">
-      {user.team ? (
+    <div className="bg-white rounded-lg p-4 gap-3 m-2 flex flex-col">
+      {user.team && !team && <Loading />}
```

Comment:
the end goal is to have it in team right? user.team value doesn't matter since team is the final output right?